### ISSUE TITLE
Is it worth installing this mod anyway?

### ISSUE THREAD

**Issue by TheCrossCarrier** [malte0811/FerriteCore#161]
Is it worth installing this mod if I allocate 12GB of memory for Minecraft (16GB on the system), and I don’t see that memory is a performance bottleneck?

**Comment by ghost** [comment]
Unless you're playing large modpacks and/or have high-quality resource/shader packs enabled, Minecraft simply doesn't need that amount of memory allocated to it. Giving too much memory to Minecraft might actually hurt performance, since it doesn't know what to do with it.
I'd recommend no more than 8GB of memory, even playing large modpacks, and then installing FerriteCore as well just to reduce the number of GCs that Java will perform.
Even if you continue using 12GB of memory, reducing the amount used will still reduce the number of GCs that Java will do, although it wouldn't be doing a lot of them anyways.

Memory will unlikely be a bottleneck anyways, but GCs might be more impactful depending on your CPU, so installing FerriteCore might have a performance impact if your CPU is weaker.

TL;DR don't allocate 12GB of memory, allocate 8GB then install FerriteCore. The performance impact of FerriteCore would be strongly dependant on your CPU.

**Comment by mainrs** [comment]
> Is it worth installing this mod if I allocate 12GB of memory for Minecraft (16GB on the system), and I don’t see that memory is a performance bottleneck?

Allocating too much memory is counterproductive. 12GB is for almost all modpacks too much. You're better of installing the mod, yes. It reduces the base line RAM usage. The less RAM Minecraft uses, the faster the garbage collection will be, generally speaking. However, you don't really need the mod if you're running lightweight mod packs. It's nice to install it for systems that don't have 16GB+ RAM available.

Garbage collection is the process that is responsible for freeing unused memory in the game. However, the more RAM you allocate, the more memory the garbage collection has to scan and decide if it can be freed. Generally speaking, during garbage collection, the game does not run; the whole process is frozen. Usually this manifests as lag spikes once the RAM is ~90% full. You can clearly see this behavior if you press F3 and monitor the RAM usage.
